src/librustc_typeck/check/mod.rs

```diff
@@ -1100,8 +1100,26 @@ fn check_fn<'a, 'tcx>(
     let outer_hir_id = fcx.tcx.hir().as_local_hir_id(outer_def_id).unwrap();
     GatherLocalsVisitor { fcx: &fcx, parent_id: outer_hir_id, }.visit_body(body);
 
+    // C-variadic fns also have a `VaList` input that's not listed in `fn_sig`
+    // (as it's created inside the body itself, not passed in from outside).
+    let maybe_va_list = if fn_sig.c_variadic {
```

Comment:
almost the same code snippet appears in src/librustc_mir/build/mod.rs

factor out into a function?

Reply:
Feel free to open an issue but I think there is a more general problem here and it also involves the whole `liberated_fn_sigs`.

That is, there should be a common way to get the "function signature as viewed from inside the body" - possibly including reading the types associated with the `hir::Body` arguments as opposed to computing them from `tcx.fn_sig(def_id)`.

Reply:
Point is, I don't want a shared functions for *just* the `c_variadic` part of this.